[export] Convert all export tests to .module()

angelayi opened the pull request:
Test Plan: CI

Differential Revision: D53075379

angelayi commented on torch/export/_unlift.py:
```diff
@@ -14,6 +15,29 @@
+def _check_input_constraints_pre_hook(self, *args, **kwargs):
+    if self._in_spec is not None:
+        try:
+            args = fx_pytree.tree_flatten_spec(
+                args, self._in_spec, exact_structural_match=True
+            )  # type: ignore[assignment]
```
@suo I had to change this back to tree_flatten_spec because it's able to handle cases where we pass in kwargs with different orders than how it was exported (ex. {"kwarg1": 1, "kwarg2":2} vs. {"kwarg2": 2, "kwarg1": 1}) . Do you have a better suggestion?

suo replied:
strictly speaking, what you described should be an error:
<img width="275" alt="image" src="https://github.com/pytorch/pytorch/assets/1617424/5dbf1121-2e6b-45c9-8fdb-73d64a8a832d">

suo replied:
either way, not supporting custom pytrees is a bigger downside than not being relaxed about dict ordering, so still not sold on tree_flatten_spec here.

suo reviewed (approved):
probably still no on tree_flatten_spec, can we rewrite the tests instead?

suo commented:
for the pytree thing https://github.com/pytorch/pytorch/pull/118505

angelayi committed:
[export] Convert all export tests to .module() (#118425)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/118425

Test Plan: CI

Reviewed By: suo

Differential Revision: D53075379
